src/components/users/UserRegisterForm.js

```diff
@@ -1,146 +1,108 @@
 
 class UserRegisterForm extends Component {
   constructor (props) {
     super(props)
     this.state = {
       snackAutoHideDuration: 2000,
       snackMessage: 'User action',
-      snackOpen: false,
-      email: null,
-      password: null,
-      passwordRepeat: null,
-      firstName: null,
-      lastName: null,
-      validFields: {
-        email: false,
-        password: false,
-        passwordRepeat: false,
-        firstName: false,
-        lastName: false
-      }
+      snackOpen: false
     }
   }
 
-  onFieldValueChange = (fieldName, fieldValue, hasError) => {
-    this.setState(prevState => ({
-      [fieldName]: fieldValue,
-      validFields: {
-        ...prevState.validFields,
-        [fieldName]: !hasError
-      }
-    }))
-  }
-
   handleRequestClose = () => {
     this.setState({ snackOpen: false })
   }
-
-  handleSubmit = event => {
-    Axios.post(URL_USERS, {
-      email: this.state.email,
-      firstName: this.state.firstName,
-      lastName: this.state.lastName,
-      password: this.state.password
-    })
-            .then(response =>
-                this.setState({
-                  snackMessage: 'User registered',
-                  snackOpen: true
-                })
-            )
-            .catch(error =>
-                this.setState({
-                  snackMessage: 'Failed to register',
-                  snackOpen: true
-                })
-            )
-  }
-
   render () {
     return (
-      <div style={main}>
+      <div>
+        <Form
+          title='User registration'
+          buttonText='Register'
+          onSubmitCallback={data => {
+            Axios.post(URL_USERS, {
+              email: data.email,
+              firstName: data.firstName,
+              lastName: data.lastName,
+              password: data.password
+            })
+                            .then(response =>
+                                this.setState({
+                                  snackMessage: 'User registered',
+                                  snackOpen: true
+                                })
+                            )
+                            .catch(error =>
+                                this.setState({
+                                  snackMessage: 'Failed to register',
+                                  snackOpen: true
+                                })
+                            )
+          }}
+                >
+          <TextField
+            name='email'
+            hintText='Email'
+            floatingLabelText='Email'
+            validationFn={value =>
+                            !emailValidator.validate(value) && 'Wrong email'}
+                    />
+          <br />
+          <TextField
+            name='password'
+            type='password'
+            hintText='Password'
+            floatingLabelText='Password'
+            validationFn={value =>
+                            value.length < 6 && 'Password to short'}
```

Comment:
Try to make those validation functions reusable, so you wouldn't repeat yourself.